addons/base_hash/models/hash_mixin.py

```diff
@@ -0,0 +1,60 @@
+from hashlib import sha256
+from json import dumps
+
+from odoo import fields, models
+
+
+class HashMixin(models.AbstractModel):
+    _name = 'hash.mixin'
+    _description = "Hash Mixin"
+
+    inalterable_hash = fields.Char(string="Inalterability Hash", readonly=True, copy=False)
+    string_to_hash = fields.Char(compute='_compute_string_to_hash', readonly=True, copy=False)
+
+    def _get_fields_used_by_hash(self):
+        """E.g.: 'create_date', 'line_ids.debit', 'done_date' """
+        raise "'_get_fields_used_by_hash' must be overriden by the inheriting class" \
+              "that uses the following '_compute_string_to_hash method'"
+
+    def _compute_hash(self, previous_hash):
```

Comment:
`_get_previous_hash` should probably declared on the mixin so that you only have to override that

Also, adding the mixin should already have an effect just by applying it; it just needs to be customizable enough so that you can do what you need to do for l10n_pt